janitor/functions/transform_columns.py

```diff
@@ -97,10 +110,11 @@ def transform_column(
         # df.
         check_column(df, dest_column_name, present=False)
 
-    if elementwise:
-        result = df[column_name].apply(function)
-    else:
-        result = function(df[column_name])
+    result = _get_transform_column_result(
+        df[column_name],
+        function,
+        elementwise,
+    )
 
     df = df.assign(**{dest_column_name: result})
```

Comment:
we can just return here, without the assignment:  ``return df.assign(**{dest_column_name: result})``